langchain-tool-retrieval.ipynb

```diff
@@ -22,329 +45,415 @@
     "from langchain.output_parsers import PydanticOutputParser\n",
     "from langchain.pydantic_v1 import BaseModel, Field\n",
     "from langchain.embeddings import SentenceTransformerEmbeddings\n",
-    "\n",
-    "from utils import map_openbb_collection_to_langchain_tools\n",
-    "\n",
+    "from langchain.output_parsers import RetryWithErrorOutputParser\n",
+    "from langchain.llms import OpenAI\n",
     "\n",
     "os.environ[\"TOKENIZERS_PARALLELISM\"] = \"False\"  # Avoid some warnings from HuggingFace\n",
+    "\n",
     "# Set up OpenAI API key\n",
-    "# os.environ[\"OPENAI_API_KEY\"] = \"\"\n",
-    "# openai.api_key = \"\""
+    "import openai\n",
+    "os.environ[\"OPENAI_API_KEY\"] = \"\"\n",
+    "openai.api_key = \"\"\n",
+    "\n",
+    "# Set up OpenBB Personal Access Token from https://my.openbb.co/app/platform/pat\n",
+    "from openbb import obb\n",
+    "from utils import map_openbb_collection_to_langchain_tools  # provides access to OpenBB Tools\n",
+    "obb.account.login(pat=\"\")"
```

Comment:
Although I suppose, it might be important for us to mention that you will need an FMP key (I think?) for these to work.

YFinance when 🥲

Reply:
The Platform team is going to suppress that output by default, so I wouldn't add any hack and just wait for that fix which should happen soon.

the .login was indeed because of the FMP key for your prompt :)